ghost/core/core/server/services/members/jobs/clean-expired-comped.js

```diff
@@ -0,0 +1,71 @@
+const {parentPort} = require('worker_threads');
+const debug = require('@tryghost/debug')('jobs:clean-expired-comped');
+const moment = require('moment');
+
+// recurring job to clean expired complimentary subscriptions
+
+// Exit early when cancelled to prevent stalling shutdown. No cleanup needed when cancelling as everything is idempotent and will pick up
+// where it left off on next run
+function cancel() {
+    if (parentPort) {
+        parentPort.postMessage('Expired complimentary subscriptions cleanup cancelled before completion');
+        parentPort.postMessage('cancelled');
+    } else {
+        setTimeout(() => {
+            process.exit(0);
+        }, 1000);
+    }
+}
+
+if (parentPort) {
+    parentPort.once('message', (message) => {
+        if (message === 'cancel') {
+            return cancel();
+        }
+    });
+}
+
+(async () => {
+    const cleanupStartDate = new Date();
+    const db = require('../../../data/db');
+    debug(`Starting cleanup of expired comp subscriptions on ${cleanupStartDate}`);
+    const expiredCompedRows = await db.knex('members_products')
+        .where('expiry_at', '<', moment.utc().startOf('day').toISOString())
```

Comment:
Is the logic here to expire products from yesterday and below or should include the ones set to expire at midnight (5 minutes before the run of this job)? in other words should it be `<` or `<=` :thinking:

Reply:
> Is the logic here to expire products from yesterday and below

It should be this, basically any sub whose expiry date is already in past and doesn't include today should be expired, as the ones that are expiring midnight means they still have access to tier till today.